src/bci_build/package/__init__.py

```diff
@@ -1031,26 +1031,50 @@ def readme_url(self) -> str:
 
     @property
     def readme(self) -> str:
-        if "README.md" in self.extra_files:
-            if isinstance(self.extra_files["README.md"], bytes):
-                return self.extra_files["README.md"].decode("utf-8")
-            return str(self.extra_files["README.md"])
+        """The contents of the image README that is refered by ``io.artifacthub.package.readme-url`` label.
+
+        The text content is obtained from :py:attr:`BaseContainerImage.extra_files` if
+        ``README.md.j2`` is defined.
+
+        If no file ``README.md.j2`` is explicitly defined in :py:attr:`BaseContainerImage.extra_files`
+        but the file is available in the image folder, assuming :py:attr:`BaseContainerImage.name`
+        as the folder name, this file going to be used.
+
+        If no proper README is found, a default text based on :py:attr:`BaseContainerImage.title`
+        and :py:attr:`BaseContainerImage.description` will be returned.
+
+        """
+        if "README.md.j2" in self.extra_files:
+            tmpl = self.extra_files["README.md.j2"]
```

Comment:
so this is the 3rd way to configure a README to use. I would prefer to keep it at the current options that exist (README.md as python rendered content and as jinja2 template). yet another way makes it just confusing rather than helpful.

Reply:
If I go with a file name that matches the image this can be removed.

Reply:
I do agree with Dirk here, I don't think that we should provide another method to obtain the readme.